src/views/visual/visual-view-graph.js

```diff
@@ -134,11 +158,12 @@ export class VisualViewGraph extends LitElement {
       ? searchTerm.split('_')[1]
       : searchTerm.replace('tib_', '');
     console.log('visual view: fetching data', this.searchItem);
-    const { graphdata, error } = await getDataForVisual({
+    let { graphdata, error } = await getDataForVisual({
       searchTerm: searchTerm,
       selected: this.selectedCollections,
       language: this.language,
     });
+    graphdata = this.preprocessGraphData(graphdata);
```

Comment:
best not use this kind of constructions that change variables in this way. Best to use const and then use another name for the output.